src/wp-includes/global-styles-and-settings.php

```diff
@@ -255,3 +255,70 @@ function ( $item ) {
 		}
 	}
 }
+
+/**
+ * Whether a theme or its parent have a theme.json file.
+ *
+ * @since 6.2.0
+ *
+ * @return bool
+ */
+function wp_theme_has_theme_json() {
+	/*
+	 * By using the 'theme_json' group, this data is marked to be non-persistent across requests.
+	 * See `wp_cache_add_non_persistent_groups` in src/wp-includes/load.php and other places.
+	 *
+	 * The rationale for this is to make sure derived data from theme.json
+	 * is always fresh from the potential modifications done via hooks
+	 * that can use dynamic data (modify the stylesheet depending on some option,
+	 * settings depending on user permissions, etc.).
+	 * See some of the existing hooks to modify theme.json behaviour:
+	 * https://make.wordpress.org/core/2022/10/10/filters-for-theme-json-data/
+	 *
+	 * A different alternative considered was to invalidate the cache upon certain
+	 * events such as options add/update/delete, user meta, etc.
+	 * It was judged not enough, hence this approach.
+	 * See https://github.com/WordPress/gutenberg/pull/45372
+	 */
+	$cache_group       = 'theme_json';
+	$cache_key         = 'wp_theme_has_theme_json';
+	$theme_has_support = wp_cache_get( $cache_key, $cache_group );
+
+	/*
+	 * $theme_has_support is stored as an int in the cache.
+	 *
+	 * The reason not to store it as a boolean is to avoid working
+	 * with the $found parameter which apparently had some issues in some implementations
+	 * https://developer.wordpress.org/reference/functions/wp_cache_get/
+	 *
+	 * Ignore cache when `WP_DEBUG` is enabled, so it doesn't interfere with the theme developers workflow.
+	 */
+	if ( ! WP_DEBUG && is_int( $theme_has_support ) ) {
+		return (bool) $theme_has_support;
+	}
+
+	// Does the theme have its own theme.json?
+	$theme_has_support = is_readable( get_stylesheet_directory() . '/theme.json' );
+
+	// Look up the parent if the child does not have a theme.json.
+	if ( ! $theme_has_support ) {
+		$theme_has_support = is_readable( get_template_directory() . '/theme.json' );
+	}
+
+	$theme_has_support = $theme_has_support ? 1 : 0;
+
+	wp_cache_set( $cache_key, $theme_has_support, $cache_group );
+
+	return (bool) $theme_has_support;
+}
+
+/**
+ * Private function to clean the caches under the theme_json group.
+ *
+ * @since 6.1.2
```

Comment:
The above code is for 6.2.0, so this function should be too.
```suggestion
 * @since 6.2.0
```

Reply:
Resolved in https://github.com/WordPress/wordpress-develop/pull/3556/commits/860418295f2765a0649d42bc15a3e5d397dcc3a2.